src/Autocomplete/assets/src/controller.ts

```diff
@@ -389,12 +393,21 @@ export default class extends Controller {
                         break;
                     }
 
+                    if (
+                        mutation.target === this.element &&
+                        mutation.attributeName.match(/data-(symfony--ux-)?autocomplete/)
+                    ) {
+                        requireReset = true;
+
+                        break;
+                    }
```

Comment:
There may be a simpler path: a stimulus value changed callback (or, really, one for each value we care about). The mutation observer is needed because we don’t have a way watch for option changes or things like the multiple attribute. But stimulus does give us a way to watch for value changes.

The only tricky part might be avoiding a “double” reset. For example, if some options are changed AND some values are changed at the same time, we only need to reset 1 time. If we use the value callbacks for one thing and the mutation observer for another, it may reset twice. Or if 3 values change, we don’t want to reset 3 times. Honestly, my idea sounds like more trouble. But since value changed callbacks are a standard solution, we should check into using them.

Reply:
That would work much better. I have added a `urlValueChanged` callback. Currently there is nothing else that triggers a reset when using remote data, so I don't think it's necessary to add a check to ensure we don't reset multiple times. When re-rendering and the url-value changes, the dropdown will only reset once.